### PR TITLE
[cdc] refactor mysql cdc action uniformly uses flatmap to process records

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Linked issue: close #xxx
refactor mysql cdc action uniformly uses flatmap to process records.

make CDC processing procedures consistent.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java [hunk 1]
```diff
@@ -284,28 +286,26 @@ public void build() throws Exception {
...
+                                tableNameConverter);
 
         String database = this.database;
         MultiTablesSinkMode mode = this.mode;
-        new FlinkCdcSyncDatabaseSinkBuilder<String>()
-                .withInput(env.fromSource(source, WatermarkStrategy.noWatermarks(), "MySQL Source"))
+        new FlinkCdcSyncDatabaseSinkBuilder<RichCdcMultiplexRecord>()
+                .withInput(
+                        env.fromSource(source, WatermarkStrategy.noWatermarks(), "MySQL Source")
+                                .flatMap(recordParser))
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java:308]: .flatMap().name("Parse")

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncTableAction.java [hunk 2]
```diff
@@ -227,23 +228,26 @@ public void build() throws Exception {
...
+
+        CdcSinkBuilder<RichCdcMultiplexRecord> sinkBuilder =
+                new CdcSinkBuilder<RichCdcMultiplexRecord>()
                         .withInput(
                                 env.fromSource(
-                                        source, WatermarkStrategy.noWatermarks(), "MySQL Source"))
+                                                source,
+                                                WatermarkStrategy.noWatermarks(),
+                                                "MySQL Source")
+                                        .flatMap(recordParser))
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncTableAction.java:250]: add .name()

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/RichCdcMultiplexRecord.java [hunk 3]
```diff
@@ -49,6 +49,10 @@ public RichCdcMultiplexRecord(
         this.cdcRecord = cdcRecord;
     }
 
+    public String databaseName() {
+        return databaseName;
+    }
+
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/RichCdcMultiplexRecord.java:55]: Seems no class call this?

Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/RichCdcMultiplexRecord.java:55]: call here:

https://github.com/apache/incubator-paimon/blob/e816c51eea232d163c4985b1a0dd94a7011164c9/paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/RichCdcMultiplexRecordEventParser.java#L87-L89

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlRecordParser.java [hunk 4]
```diff
@@ -211,87 +168,83 @@ public List<DataField> parseSchemaChange() {
...
+                    "Didn't find primary keys from MySQL DDL for table '{}'. "
+                            + "This table won't be synchronized.",
+                    currentTable);
+            nonPkTables.add(currentTable);
             return Collections.emptyList();
         }
 
-        Optional<Schema> schema = schemaBuilder.build(tableChange);
-        return schema.get().fields();
+        nonPkTables.remove(currentTable);
```
[the remark this answers is not in the document]
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlRecordParser.java]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java (hunk 1 above)
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java]: done.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**